<?xml version="1.0"?>
<launch>
  <arg name="gdb_debug" default="false"/>
  <group ns="msckf">
    <param name="data_set_path" value="/home/ken/euroc_datasets/MH_03_medium/mav0"/>
    <param name="stand_still_end" value="1403637149088318946."/>

    <param name="feature_cov" value="3"/>

    <param name="imu_vars/w_var" value="1e-4"/>
    <param name="imu_vars/dbg_var" value="3.6733e-5"/>
    <param name="imu_vars/a_var" value="1e-2"/>
    <param name="imu_vars/dba_var" value="7e-2"/>

    <param name="imu_covars/p_init" value="1e-12"/>
    <param name="imu_covars/q_init" value="1e-5"/>
    <param name="imu_covars/v_init" value="1e-2"/>
    <param name="imu_covars/ba_init" value="1e-2"/>
    <param name="imu_covars/bg_init" value="1e-2"/>

    <param name="max_gn_cost_norm" value="7"/>
    <param name="translation_threshold" value="0.005"/><!--Max translation from marginalizing state-->

    <param name="keyframe_transl_dist" value="0.5"/>
    <param name="keyframe_rot_dist" value="0.5"/>

    <param name="min_track_length" value="5"/>
    <param name="max_track_length" value="1000"/>
    <param name="max_cam_states" value="20"/>

    <param name="ransac_threshold" value="0.0000002"/>

    <param name="n_grid_rows" value="8"/>
    <param name="n_grid_cols" value="8"/>

    <node pkg="msckf_mono" name="asl_msckf" type="asl_msckf" unless="$(arg gdb_debug)" output="screen"/>
    <node pkg="msckf_mono" name="asl_msckf" type="asl_msckf" launch-prefix="xterm -e gdb --args" if="$(arg gdb_debug)"/>

    <node pkg="rviz" name="msckf_rviz" type="rviz" args="-d $(find msckf_mono)/launch/asl_msckf.rviz"/>

  </group>
</launch>
